<?xml version="1.0"?>

<launch>
  <arg name="mav_name" default="m100"/>
  <arg name="namespace" default="$(arg mav_name)"/>
  <arg name="model" default="$(find uav_simulator)/urdf/$(arg mav_name)_base.xacro"/>
  <arg name="tf_prefix" default="$(optenv ROS_NAMESPACE)"/>
  <arg name="x" default="0.0"/>
  <arg name="y" default="0.0"/>
  <arg name="z" default="0.44"/>
  
  <!-- send the robot XML to param server -->
  <param name="robot_description" command="
    $(find xacro)/xacro '$(arg model)'
    mav_name:=$(arg mav_name)
    namespace:=$(arg namespace)"
  />
  <param name="tf_prefix" type="string" value="$(arg tf_prefix)" />

  <!-- push robot_description to factory and spawn robot in gazebo -->
  <node name="spawn_$(arg namespace)" pkg="gazebo_ros" type="spawn_model"
   args="-param robot_description
         -urdf
         -x $(arg x)
         -y $(arg y)
         -z $(arg z)
         -model $(arg namespace)"
   respawn="false" output="screen">
  </node>
</launch>
